### ISSUE TITLE
 Invite message with apostrophe is converted to uni code in invite SMS

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#20081]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app
2. Open settings
3. Open workspaces
4. Open any workspace
5. Open manage members
6. Click on invite
7. Select your contact number user (user which has contact number as primary contact and that number is available with you to receive text msg) and continue
8. Write any invite message with apostrophe eg: This is test's message
9. Click on invite
10. On your device which has that contact number, observe the message received

## Expected Result:
 App should not convert apostrophe into uni code in invite SMS

## Actual Result:
App converts apostrophe into uni code in invite SMS

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.22.0

**Reproducible in staging?:** yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/1af43204-6ccb-48f6-83c2-0cc3aca44de0

https://github.com/Expensify/App/assets/93399543/01967ce2-d3e7-4920-94b7-b5233cc53694

![Screenshot_20230602_132245_Messages](https://github.com/Expensify/App/assets/93399543/cd64c129-5220-47e1-8aec-4c74e5534df9)

**Expensify/Expensify Issue URL:**

**Issue reported by:** @dhanashree-sawant

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684770159646489

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by therealsujitk** [comment]
The welcome message escaped for HTML emails - #16552, I guess it doesn't make sense to escape SMS messages. To resolve this the message will have to be unescaped internally before sending the SMS.

**Comment by maddylewis** [comment]
i can repro this: 

![IMG_1530](https://github.com/Expensify/App/assets/38016013/8f08cefb-89e1-46c8-9342-d1e87bd9f8a7)

**Comment by maddylewis** [comment]
@cristipaval - just confirming that this is an `Internal` issue before adding that label - lmk!

**Comment by cristipaval** [comment]
Yes, this seems to be internal. I can fix it early next week when I am back home. I am currently having some roaming issue.

**Comment by cristipaval** [comment]
I investigated this a bit and it seems that it could be made external. The App sends `#x27;` instead of the single quote.

**Comment by akinwale** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The app converts apostrophes into Unicode in invite messages sent by SMS.

### What is the root cause of that problem?
The `_.escape` method is used on the `welcomeNote` when executing the `AddMembersToWorkspace` action to `API.write`. Based on the [documentation](https://underscorejs.org/#unescape), this converts apostrophes to `&#x27;`.

### What changes do you think we should make in order to solve the problem?
We can use one of the following solutions to solve the problem.

Solution 1
Use a different escape method which does not encode the apostrophe.

Solution 1
Do not escape the welcomeNote. This means that for email messages (where HTML is inserted), the backend will have to escape the welcomeNote.

Solution 2
Allow passing a different parameter called smsWelcomeNote to the API. This will accept the unescaped welcomeNote string.


### What alternative solutions did you explore? (Optional)
None.

**Comment by maddylewis** [comment]
@0xmiroslav - what do you think of @akinwale's proposal above? whenever you have a chance, give that look and provide an update on your recommended next steps. thanks!

**Comment by cristipaval** [comment]
@0xmiroslav has to review the proposal

**Comment by akinwale** [comment]
@0xmiroslav Just checking in here. Did you get a chance to review my [my proposal](https://github.com/Expensify/App/issues/20081#issuecomment-1588398016) yet?

**Comment by maddylewis** [comment]
bumping @0xmiroslav to review @akinwale's proposal so that we can either move forward with that one or find a different solution - thanks!

**Comment by maddylewis** [comment]
@cristipaval - whenever you have a sec, let me know your thoughts on [this comment](https://github.com/Expensify/App/issues/20081#issuecomment-1603095263):

> @akinwale's Solution 1 or 2 will work. @cristipaval which solution is better? I vote 2. since it's edge case that welcome message has html characters.
For solution 2, conditional logic should be done in backend as the logic of whether to send email or SMS is in backend.

**Comment by cristipaval** [comment]
Hmm.. I will discuss this internally with the team.

**Comment by cristipaval** [comment]
@0xmiroslav @akinwale I like the second solution. I am going to assign @akinwale, but his PR will be on hold until the backend change will get deployed into production. I'll work on the backend change soon.

**Comment by akinwale** [comment]
@cristipaval @0xmiroslav The PR is ready for review. Thanks.

**Comment by cristipaval** [comment]
After spending some time, it looks like the cleanest way to fix this is to let it be sent escaped by the front-end and unescape on the backend if the user is notified via SMS. Sorry about changing my mind. I'll close your PR @akinwale. 

@maddylewis do we have a bonus for the investigation that @akinwale did?
It was like this:
1. @0xmiroslav and I considered @akinwale solution good. 
2. The solution was supposed to have 2 PRs, his App PR and one Web-E PR.
3. @akinwale opened the App PR
4. I started to work on Web-E PR and realized that the best solution is to fix it in backend only
5. I closed @akinwale's PR

**Comment by maddylewis** [comment]
thanks for the update @cristipaval. yep, i think it's fair to still issue payment to @0xmiroslav and @akinwale for the work they did on this one. 

> We can still help testing after backend PR deployed to staging/production.

with that ^^^ in mind, do we think $500 each makes sense for this one?

**Comment by cristipaval** [comment]
yes @maddylewis, that's great! Thank you! 🙇

**Comment by cristipaval** [comment]
I'll let @akinwale and @0xmiroslav know when the Web-PR is in staging and can be tested.

@maddylewis , I'll also ping you when everything is done and the payments can be issued.

**Comment by maddylewis** [comment]
sounds good - ty @cristipaval!

**Comment by cristipaval** [comment]
After internal discussions, we ended up deciding to not escape the `welcomeNote` in the front end. This means that I reopened the @akinwale [PR](https://github.com/Expensify/App/pull/21912), which is held on my [Web-E PR](https://github.com/Expensify/Web-Expensify/pull/38012). 
@maddylewis If everything goes as per the latest plan, we will have to do full payments to @akinwale and @0xmiroslav

**Comment by maddylewis** [comment]
noting payments: 
- [ ] $250 payment to @dhanashree-sawant
- [ ] $1000 to @0xmiroslav 
- [ ] $1000 to @akinwale

**Comment by akinwale** [comment]
@maddylewis I assume this would be eligible for the 50% speed bonus since my PR was ready same day I was assigned, and there have been no changes since then. We've been waiting for the backend PR to get merged.

**Comment by maddylewis** [comment]
updated amounts!

**Comment by cristipaval** [comment]
I was ooo last week and that's why the [Web-E PR](https://github.com/Expensify/Web-Expensify/pull/38012) took so long. I am waiting for the (hopefully) last round of reviews.

**Comment by cristipaval** [comment]
21 days elapsed due to my ooo time and internal discussions. No penalty for this one, @maddylewis.
- @dhanashree-sawant should be paid for the reporting bonus
- @akinwale for the solution
- @0xmiroslav as C+

**Comment by cristipaval** [comment]
@0xmiroslav can you please check the comment above?

**Comment by cristipaval** [comment]
Waiting for @0xmiroslav

**Comment by maddylewis** [comment]
apologies! i issued a couple of payments early:
- [ ] @dhanashree-sawant should be paid for the reporting bonus - paid early 
- [ ] @akinwale for the solution - paid early
- [ ] @0xmiroslav as C+ - waiting for Miroslav to accept the offer in Upwork and then i will process that payment

**Comment by maddylewis** [comment]
@0xmiroslav - should i hire `Volodomyr` to this issue?

**Comment by maddylewis** [comment]
okay, here is the updated job post - https://www.upwork.com/jobs/~0163e9e66ca38f82a1

im still a little unclear how to correctly send the payment to you. i will ask internally and then follow up.

**Comment by maddylewis** [comment]
i just need to pay @0xmiroslav and then i can close this out. i confirmed how to do that on Friday so doing now!

**Comment by maddylewis** [comment]
sent offer to Volodymyr

**Comment by maddylewis** [comment]
everyone is paid! https://github.com/Expensify/App/issues/20081#issuecomment-1640381493
